packages/backend/src/scheduler/SchedulerTask.ts

```diff
@@ -746,6 +758,180 @@ export const sendEmailNotification = async (
     }
 };
 
+export const sendGdriveNotification = async (
+    jobId: string,
+    notification: GdriveNotificationPayload,
+) => {
+    const { schedulerUuid, schedulerGdriveTargetUuid, scheduledTime } =
+        notification;
+
+    analytics.track({
+        event: 'scheduler_notification_job.started',
+        anonymousId: LightdashAnalytics.anonymousId,
+        properties: {
+            jobId,
+            schedulerId: schedulerUuid,
+            schedulerTargetId: schedulerGdriveTargetUuid,
+            type: 'gdrive',
+        },
+    });
+
+    try {
+        const scheduler =
+            await schedulerService.schedulerModel.getSchedulerAndTargets(
+                schedulerUuid,
+            );
+        const { format, savedChartUuid, dashboardUuid, name, targets } =
+            scheduler;
+
+        const target = targets
+            .filter(isGdriveTarget)
+            .find(
+                (t) =>
+                    t.schedulerGdriveTargetUuid === schedulerGdriveTargetUuid,
+            );
+
+        if (!target) {
+            throw new Error('Gdrive destination not found');
+        }
+        const { gdriveId } = target;
+        schedulerService.logSchedulerJob({
+            task: 'sendGdriveNotification',
+            schedulerUuid,
+            jobId,
+            jobGroup: notification.jobGroup,
+            scheduledTime,
+            target: gdriveId,
+            targetType: 'gdrive',
+            status: SchedulerJobStatus.STARTED,
+        });
+
+        // Backwards compatibility for old scheduled deliveries
+        const { url, details, pageType, imageUrl, csvUrl, csvUrls } =
+            notification.page ??
+            (await getNotificationPageData(scheduler, jobId));
+        const schedulerUrl = `${url}?scheduler_uuid=${schedulerUuid}`;
+
+        if (format === SchedulerFormat.IMAGE) {
+            throw new Error('Not implemented');
+        } else if (savedChartUuid) {
+            if (csvUrl === undefined) {
+                throw new Error('Missing CSV URL');
+            }
+
+            const csvContent = await fsPromise.readFile(csvUrl.localPath, {
+                encoding: 'utf-8',
+            });
+
+            const refreshToken = await userService.getRefreshToken(
+                scheduler.createdBy,
+            );
+            await googleDriveClient.appendToSheet(
+                refreshToken,
+                gdriveId,
+                csvContent,
+            );
+        } else if (dashboardUuid) {
+            throw new Error('Not implemented');
```

Comment:
This will be implemented next in https://github.com/lightdash/lightdash/issues/6673